src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -307,6 +307,7 @@ public async Task<UpsertOutcome> UpsertAsync(ResourceWrapperOperation resource,
                 // The backwards compatibility behavior of Stu3 is to return 412 Precondition Failed instead of a 400 Client Error
                 if (_modelInfoProvider.Version == FhirSpecification.Stu3)
                 {
+                    _logger.LogInformation("PreconditionFailed: IfMatchHeaderRequiredForResource");
```

Comment:
Can we move this outside of inner if, so it will be logged for both fhir versions?

Reply:
Thanks for the suggestion. I added special logs for both cases, given that for STU3 a different type of exception is raised.